### PR TITLE
added filter for statement statistics that contain null entries

### PR DESCRIPTION
**Pull request by SamNelson**
This should fix [issue #39](https://github.com/istanbuljs/istanbuljs/issues/39)

### CODE REVIEW

File: packages/istanbul-reports/lib/html/annotator.js [hunk 1]
```diff
@@ -41,9 +41,11 @@ function annotateLines(fileCoverage, structuredText) {
 }
 
 function annotateStatements(fileCoverage, structuredText) {
-    var statementStats = fileCoverage.s,
-        statementMeta = fileCoverage.statementMap;
-    Object.keys(statementStats).forEach(function (stName) {
+    var statementStats = Object.keys(fileCoverage.s).filter(function (stName) {
+      return fileCoverage.s[stName] !== null;
```
Comment (JaKXz) [inline packages/istanbul-reports/lib/html/annotator.js]: would it be overly poor form to use `!= null` to capture any ones that are `undefined` as well? or would it be overkill to do `&& !== undefined`?

File: packages/istanbul-reports/lib/html/annotator.js [hunk 2]
```diff
@@ -41,9 +41,11 @@ function annotateLines(fileCoverage, structuredText) {
...
+    }),
+    statementMeta = fileCoverage.statementMap;
```
Comment (JaKXz) [inline packages/istanbul-reports/lib/html/annotator.js]: I think this should be a `var statementMeta` line, it's kind of hard to grok imho.

**Comment by bcoe** [comment]
@JaKXz I'm pretty sure that in normal circumstances, getting a `null` statement should be impossible;

@SamNelson are you able to share the codebase that you were bumping into issues on? I think if we switch the order of `istanbul` and `rewire` things might start working for you -- and we won't need this fix.

**Comment by JaKXz** [comment]
@bcoe wouldn't it be good to have this passing anyways? in case there are other scenarios where there are null entries in statement statistics?

**Comment by bcoe** [comment]
@JaKXz the problem is it would result in misreporting test coverage information -- except in bizarre situations, like the interaction we're seeing between `babel-plugin-rewire` and `babel-plugin-istanbul`, it should be impossible to have a `null` in a coverage map. If we'd been hiding this edge-case, we never would have found the bad interaction between these two libraries that was dropping coverage information on the ground.

**Comment by bcoe** [comment]
@SamNelson please feel free to reopen this discussion if you're still bumping into issues 👍 we've rolled out a few changes to nyc/istanbul-lib-instrument that I think should solve your problem.

**Pull request closed without merging**